fe/src/main/java/org/apache/doris/analysis/StmtRewriter.java

```diff
@@ -43,47 +45,52 @@
      * Rewrite the statement of an analysis result. The unanalyzed rewritten
      * statement is returned.
      */
-    public static void rewrite(Analyzer analyzer, StatementBase parsedStmt)
+    public static StatementBase rewrite(Analyzer analyzer, StatementBase parsedStmt)
             throws AnalysisException {
         if (parsedStmt instanceof QueryStmt) {
             QueryStmt analyzedStmt = (QueryStmt) parsedStmt;
             Preconditions.checkNotNull(analyzedStmt.analyzer);
-            rewriteQueryStatement(analyzedStmt, analyzer);
+            parsedStmt = rewriteQueryStatement(analyzedStmt, analyzer);
```

Comment:
Better to use a new local variable and return it. Not to return a `parameter` variable.
And I think you can just return here.

```
return rewriteQueryStatement(analyzedStmt, analyzer);
```